OMPlot/OMPlotGUI/PlotWindow.cpp

```diff
@@ -41,9 +41,15 @@
 
 using namespace OMPlot;
 
-PlotWindow::PlotWindow(QStringList arguments, QWidget *parent)
-  : QMainWindow(parent)
+PlotWindow::PlotWindow(QStringList arguments, QWidget *parent, bool isInteractiveSimulation, QToolButton *pStartSimulation,
+                       QToolButton *pPauseSimulation, QComboBox *pSimulationSpeed)
+  : QMainWindow(parent), mIsInteractiveSimulation(isInteractiveSimulation)
```

Comment:
It was to keep track of the targeted simulation to begin with, but I agree that it is weird and should be done as you said.